File: algorithm/miniMaxSum/answer.js
// Answer 1
function miniMaxSumOld(arr) {
  const min = Math.min(...arr);
  const max = Math.max(...arr);

  let maxSum = 0;
  let minSum = 0;
  for (let i = 0; i < arr.length; i++) {
    if (value !== max) minSum += arr[i];
    if (value !== min) maxSum += arr[i];
  }

  console.log(`${minSum} ${maxSum}`);
}

// Answer 2
function miniMaxSum(arr) {
  let sums = [];

  for (let i = 0; i < arr.length; i++) {
    const total = arr.reduce((acc, cur) => {
      if (cur !== arr[i]) return acc + cur;
      else return acc;
    }, 0);
    sums.push(total);
  }

  console.log(`${Math.min(...sums)} ${Math.max(...sums)}`);
}

File: algorithm/problem/miniMaxSum/answer.py
# Anwser 1
def miniMaxSumOne(arr):
  minNum = min(arr)
  maxNum = max(arr)

  minSum = 0
  maxSum = 0
  for i in range(0, len(arr)):
    value = arr[i]
    if value != minNum:
      maxSum += value
    if value != maxNum:
      minSum += value

  print(str(minSum) + " " + str(maxSum))

# Anwser 2
def miniMaxSumTwo(arr):
  sums = []

  for i in range(0, len(arr)):
    cloneArr = list.copy(arr)
    del cloneArr[i]
    sums.append(sum(cloneArr))

  minSum = min(sums)
  maxSum = max(sums)
  print(str(minSum) + " " + str(maxSum))
